modules/core/src/v2/coins/xlm.ts

```diff
@@ -1,23 +1,148 @@
 import * as _ from 'lodash';
-import { BigNumber } from 'bignumber.js';
+import * as bitcoin from 'bitgo-utxo-lib';
 import * as querystring from 'querystring';
 import * as url from 'url';
 import * as Bluebird from 'bluebird';
-import * as bitcoin from 'bitgo-utxo-lib';
-const co = Bluebird.coroutine;
 import * as request from 'superagent';
 import * as stellar from 'stellar-sdk';
+import { BigNumber } from 'bignumber.js';
 
-import { BaseCoin } from '../baseCoin';
 import { Ed25519KeyDeriver } from '../keyDeriver';
 import * as config from '../../config';
 import * as common from '../../common';
+import {
+  InvalidAddressError,
+  InvalidMemoIdError,
+  KeyRecoveryServiceError,
+  UnexpectedAddressError
+} from '../../errors';
+import { BaseCoin } from '../baseCoin';
+import { NodeCallback } from '../types';
+import { Wallet } from '../wallet';
 
-const maxMemoId = '0xFFFFFFFFFFFFFFFF'; // max unsigned 64-bit number = 18446744073709551615
+const co = Bluebird.coroutine;
+
+interface KeyPair {
+  pub?: string;
+  prv: string;
+}
+
+interface AddressDetails {
+  address: string;
+  memoId?: string;
+}
+
+interface Memo {
+  type: stellar.MemoType;
```

Comment:
what is this type? I think it should be `string`

Reply:
It's a type exported by the stellar sdk.

Here's the definition:
```
enum MemoType
{
    MEMO_NONE = 0,
    MEMO_TEXT = 1,
    MEMO_ID = 2,
    MEMO_HASH = 3,
    MEMO_RETURN = 4
};
```